src/kernel/sporadic.c

```diff
@@ -227,6 +222,51 @@ void refill_update(sched_context_t *sc, ticks_t new_period, ticks_t new_budget,
     REFILL_SANITY_CHECK(sc, new_budget);
 }
 
+void refill_budget_check_round_robin(ticks_t usage)
```

Comment:
why is there a refill check for round robin? RR threads should be much simpler than this, where tail = consumed so far, head = what's left of this timeslice. Finish timeslice = tail = 0, head = budget.

Reply:
I will look into simplifying RR independently of sporadic. I think because we had to prove a large number of invariants about all SCs we lost track of what was only particular to sporadic SCs.